pytorch_lightning/trainer/training_loop.py

```diff
@@ -555,7 +555,8 @@ def check_checkpoint_callback(self, should_check_val):
             [c.on_validation_end(self, self.get_model()) for c in checkpoint_callbacks]
 
     def update_train_loop_lr_schedulers(self):
-        if (self.batch_idx + 1) % self.accumulate_grad_batches == 0:
+        if ((self.batch_idx + 1) % self.accumulate_grad_batches == 0
+                or self.num_training_batches == (self.batch_idx + 1)):
```

Comment:
```suggestion
                or (self.batch_idx + 1) == self.num_training_batches):
```
for alignment with first line / readibility